cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/IndexCreatorTest.java

```diff
@@ -31,124 +38,126 @@
 
     @Test
     public void emptyIndexList() {
-        Map<String, Map<String, Object>> indexes = im.listIndexes();
+        List<Index> indexes = im.listIndexes();
         assertThat(indexes, is(notNullValue()));
         assertThat(indexes.isEmpty(), is(true));
     }
 
     @Test
-    public void preconditionsToCreatingIndexes() {
+    public void preconditionsToCreatingIndexes() throws QueryException {
```

Comment:
I think this test needs to be broken up into its individual tests using expected class to make sure the exceptions get thrown.